app/controllers/questions.rb

```diff
@@ -1,16 +1,37 @@
-get '/questions/new' do
+get '/categories/:cid/questions/new' do
+	if logged_in?
+		@category = Category.find(params[:cid])
+		@all_categories = Category.order(:title)
+		erb :'/questions/new'
+	else 
+		redirect '/users/login'
```

Comment:
@kmbradshaw: it’s actually kind of moot because the button should not be visible unless they’re logged in